pypeit/spec2dobj.py

```diff
@@ -531,28 +579,9 @@ def build_primary_hdr(self, raw_header, spectrograph, master_key_dict=None, mast
         hdr['PYP_SPEC'] = spectrograph.name
         hdr['DATE-RDX'] = str(datetime.date.today().strftime('%Y-%m-%d'))
 
-        # MasterFrame info
-        # TODO -- Should this be in the header of the individual HDUs ?
-        if master_key_dict is not None:
-            if 'bias' in master_key_dict.keys():
-                hdr['BIASMKEY'] = master_key_dict['bias']
-            if 'arc' in master_key_dict.keys():
-                hdr['ARCMKEY'] = master_key_dict['arc']
-            if 'trace' in master_key_dict.keys():
-                hdr['TRACMKEY'] = master_key_dict['trace']
-            if 'flat' in master_key_dict.keys():
-                hdr['FLATMKEY'] = master_key_dict['flat']
-
-        # Processing steps
-        # TODO: Assumes processing steps for all detectors are the same...  Does
-        # this matter?
-        det = self.detectors[0]
-        if self[det].process_steps is not None:
-            hdr['PROCSTEP'] = (','.join(self[det].process_steps), 'Completed reduction steps')
-
         # Some paths
-        if master_dir is not None:
-            hdr['PYPMFDIR'] = str(master_dir)
+        if calib_dir is not None:
```

Comment:
I presume these are absolute paths. There is a problem with the visualization or chk scripts if the user moves those directories. My feeling is that these scripts should be run from the reduction directory with the pypeit file and that Calibrations and Science are standard paths, and we can specify everything relative to that directory. I'm not sure this impacts what is being set here. Probably best to write absolute paths to the headers, but then thing about how to modify the chk and visualization scripts to use relative paths when needed.

Reply:
I agree! Relative paths are better. While I was testing, I changed the name of the folder and when I went back to check the results or to re-run the reduction (using the existing calibrations), I was not able to do so because of the absolute paths.

Reply:
Making a coherent choice of relative vs. absolute paths and then implementing that is something for a different PR.